### PR TITLE
Google OAuth2 login added

### PR DESCRIPTION
**Pull request by YouruiR**
## Overview

Issue Type

- [ ] Bug
- [x] Feature
- [ ] Tech Debt

Description
Users can now login to the site with their Google accounts.

Removed old API Key files and moved API Keys to a .env file.

Ticket Item

FEATURE#2: DB Spike #1

Steps to Validate Feature

1. Start the frontend and backend server
2. Go to 'http://127.0.0.1:5173'
3. Click on 'Login' at the top right of the nav bar
4. Choose your google account to login with
5. You should be redirected back to the home page. Address bar should now contain your OAuth access token.

Screenshots & Videos

Screenshot 2023-04-10 at 12 55 35 AM

Screenshot 2023-04-10 at 12 57 43 AM

Additional context
Will need to touch base with the group on creating the .env files to ensure team still has access to the API's in their own tests.

### CODE REVIEW

File: client/vite.config.js [hunk 1]
```diff
@@ -8,5 +8,8 @@ export default defineConfig({
       '/api': 'http://localhost:4000/'
     }
   },
+  // define: {
```
Comment (EstherNahm) [inline client/vite.config.js]: Do we need to keep this commented in or is this safe to remove/clean up?

File: server/server.js [hunk 2]
```diff
@@ -7,6 +8,14 @@ app.use(express.json());
 app.use('/api/trails', bikeTrailsRouter);
 app.use('/api/moreInfo', bikeTrailInfoRouter);
 
+// app.get('/', (req, res) => {
```
Comment (EstherNahm) [inline server/server.js:11]: Remove?

File: server/server.js [hunk 3]
```diff
@@ -7,6 +8,14 @@ app.use(express.json());
...
+
+// })
+
+app.get('/api', (req, res) => {
```
Comment (EstherNahm) [inline server/server.js]: This mock endpoint can be removed now, right?

File: client/vite.config.js (hunk 1 above)
Comment (YouruiR) [reply client/vite.config.js]: yup, this can be removed.

File: server/server.js (hunk 2 above)
Comment (YouruiR) [reply server/server.js:11]: I commented that out as a reminder to talk to you guys about possibly making a landing page. Right now, the OAuth just redirects you back to the home page.

File: server/server.js (hunk 3 above)
Comment (YouruiR) [reply server/server.js]: correct, this can be taken out

**Comment by EstherNahm** [comment]
Thinking we should wait to merge this PR until we've had a chance to touch base re:setting up env file - thoughts?

**Comment by YouruiR** [comment]
I agree, I can demo the OAuth and show how to setup the .env files when we meet up in class

**Review by EstherNahm — APPROVED** [review]

**Review by jennyschmalz — APPROVED** [review]

**Pull request merged**